db/database.go

```diff
@@ -1153,11 +1172,15 @@ outerLoop:
 				startKey = queryRow.Name
 			}
 			if len(rowID) < lenDbUserPrefix && len(rowID) < lenDbRolePrefix {
+				// make sure to increment the resultCount to not skip the next row if the results are
+				// _sync:user:alice and we are looking for _sync:user:verylongdbname:alice
+				if resultCount == 0 && anyPrincipalRe.MatchString(rowID) {
+					resultCount++
```

Comment:
I'm not clear why we need to increment resultCount here and not on line 1185.

Reply:
I kept this code the same, I'm not clear on this either. 

The reason this logic is different is that we only skip the values if it is _not_ a principal document, and I'm preserving that. In the legacy logic:

For a user/rowID like: `_sync:user:alice`:

`len(rowID) < lenDbUserPrefix && len(rowID) < lenDbRolePrefix` -> false
`if !isDbUser && !isDbRole` -> false

For a case where we have:

`_sync:user:alice`
`_sync:user:namedcollectiondb:alice`

`len(rowID) < lenDbUserPrefix && len(rowID) < lenDbRolePrefix` -> true
`if !isDbUser && !isDbRole` -> false